docs/misc/str-builder-api.rst

```diff
@@ -1,443 +0,0 @@
-bytes/str building API
-=======================
```

Comment:
Maybe move this into a "design document" section?

Reply:
I've linked it to the relevant GitHub issue. I think that having design documents in the "main" user documentation is not good, because:
* it can be misleading and create false expectations
* we should primarily use GitHub issues (or something else, but a single location -- I haven't read it, but I suspect that `str-builder-api.rst` is not up-to-date, anyway)